File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/events/EventFrameListener.java
package com.abubusoft.xenon.animations.events;

import com.abubusoft.xenon.animations.KeyFrame;

public interface EventFrameListener<K extends KeyFrame> {

	/**
	 * Evento relativo al cambio del key frame. Siamo all'inizio del frame.  
	 * 
	 * @param currentFrame
	 * 		key frame attualmente usato. 
	 */
	void onFrameBegin(K currentFrame);
	
	/**
	 * fine frame
	 * @param currentFrame 
	 */
	void onFrameEnd(K currentFrame);

}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/events/EventFrameListener.kt
package com.abubusoft.xenon.animations.events

import com.abubusoft.xenon.animations.KeyFrame

interface EventFrameListener<K : KeyFrame?> {
    /**
     * Evento relativo al cambio del key frame. Siamo all'inizio del frame.
     *
     * @param currentFrame
     * key frame attualmente usato.
     */
    fun onFrameBegin(currentFrame: K)

    /**
     * fine frame
     * @param currentFrame
     */
    fun onFrameEnd(currentFrame: K)
}